### PR TITLE
Detect ConfigureAwait problems on all main projects

### PR DESCRIPTION
**Pull request by jskeet**
Part of #1057.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.csproj [hunk 1]
```diff
@@ -21,10 +21,18 @@
...
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="ConfigureAwaitChecker.Analyzer" Version="1.0.0-beta4">
+      <PrivateAssets>All</PrivateAssets>
+    </PackageReference>
     <PackageReference Include="Google.Api.Gax.Rest" Version="2.0.0-beta01" />
     <PackageReference Include="Google.Apis.Bigquery.v2" Version="1.26.2.861" />
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="ConfigureAwaitChecker.Analyzer" Version="1.0.0-beta4">
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.csproj]: Is this meant to be in this file twice?

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.DownloadObject.cs [hunk 2]
```diff
@@ -117,29 +117,30 @@ private string GetBaseUri(Object source)
...
-            {
-                downloader.ProgressChanged += progress.Report;
-                // Avoid reporting progress synchronously in the original call.
-                await Task.Yield();
-                progress.Report(InitialDownloadProgress.Instance);
-            }
-            var result = await downloader.DownloadAsync(uri, destination, cancellationToken).ConfigureAwait(false);
-            if (result.Status == DownloadStatus.Failed)
+            IProgress<IDownloadProgress> progress) =>
+            // Use Task.Run to prevent reporting rogress synchronously in the original call.
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.DownloadObject.cs]: *p*rogress

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.csproj (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.csproj]: Gah, no - that was once manually and once automatically. Will fix, thanks.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.DownloadObject.cs (hunk 2 above)
Comment (jskeet) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.DownloadObject.cs]: Done.

**Comment by jskeet** [comment]
Looks like I'll need to adjust the code health tester too...

**Pull request merged**